integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -387,6 +389,90 @@ fn next_block_has_new_shard_layout(epoch_manager: Arc<dyn EpochManagerAdapter>,
         && shard_layout != next_shard_layout
 }
 
+/// Asserts that for each child shard:
+/// MemTrie, FlatState and DiskTrie all contain the same key-value pairs.
+fn assert_state_sanity_for_children_shard(parent_shard_uid: ShardUId, client: &Client) {
+    let final_head = client.chain.final_head().unwrap();
+
+    for child_shard_uid in client
+        .epoch_manager
+        .get_shard_layout(&final_head.epoch_id)
+        .unwrap()
+        .get_children_shards_uids(parent_shard_uid.shard_id())
+        .unwrap()
+    {
+        let state_root = *client
+            .chain
+            .get_chunk_extra(&final_head.prev_block_hash, &child_shard_uid)
+            .unwrap()
+            .state_root();
+
+        // Here memtries will be used as long as client has memtries enabled.
+        let memtrie = client
+            .runtime_adapter
+            .get_trie_for_shard(
+                child_shard_uid.shard_id(),
+                &final_head.prev_block_hash,
+                state_root,
+                false,
+            )
+            .unwrap();
+        assert!(memtrie.has_memtries());
+        let memtrie_state =
+            memtrie.lock_for_iter().iter().unwrap().collect::<Result<HashSet<_>, _>>().unwrap();
+
+        // To get a view on disk tries we can leverage the fact that get_view_trie_for_shard() never
+        // uses memtries.
+        let trie = client
+            .runtime_adapter
+            .get_view_trie_for_shard(
+                child_shard_uid.shard_id(),
+                &final_head.prev_block_hash,
+                state_root,
+            )
+            .unwrap();
+        assert!(!trie.has_memtries());
```

Comment:
Thanks for adding this check! I was concerned some future change to view_trie might break this test but having the assert assures it would work!